src/libstd/collections/hash/map.rs

```diff
@@ -1664,8 +1665,18 @@ impl RandomState {
     #[allow(deprecated)] // rand
     #[stable(feature = "hashmap_build_hasher", since = "1.7.0")]
     pub fn new() -> RandomState {
-        let mut r = rand::thread_rng();
-        RandomState { k0: r.gen(), k1: r.gen() }
+        thread_local!(static KEYS: (u64, u64) = {
+            // get some random bytes from the OS
+            let mut bytes = [0_u8; 16];
+            internal_rand::fill_bytes(&mut bytes);
+
+            let keys: (u64, u64) = unsafe { mem::transmute(bytes) };
```

Comment:
A transmute like this should work fine: it's reinterpreting the value, not the memory location. That is, `let out = transmute::<A, B>(x)` is/should be equivalent to (in C)

```C
B out;
memcpy(&out, &x, sizeof(B))
```

It would definitely be problematic if it was `&[u8; 16]` to `&(u64, u64)`.

(If the above is not the case, it seems to me like it is an unnecessary footgun.)